relay-server/src/services/project.rs

```diff
@@ -1136,23 +1005,40 @@ impl Project {
             }
         }
 
+        let quotas = project_state.config.quotas.clone();
+        let buckets = match MetricsLimiter::create(buckets, quotas, scoping) {
+            Ok(mut bucket_limiter) => {
+                bucket_limiter.enforce_limits(current_limits, metric_outcomes, outcome_aggregator);
+                bucket_limiter.into_buckets()
+            }
+            Err(buckets) => buckets,
+        };
+
         if buckets.is_empty() {
-            return None;
+            return CheckBuckets::Dropped;
         }
 
-        let project_metrics = ProjectMetrics {
-            buckets,
+        CheckBuckets::Checked {
+            scoping,
             project_state,
-        };
-
-        Some((scoping, project_metrics))
+            buckets,
+        }
     }
 }
 
+#[derive(Debug)]
+pub enum CheckBuckets {
+    NoProject(Vec<Bucket>),
+    Checked {
```

Comment:
nit and personal opinion, feel free to ignore:
```suggestion
pub enum CheckedBuckets {
    NoProject(Vec<Bucket>),
    Ok {
```